src/mono/wasm/runtime/startup.ts

```diff
@@ -237,8 +243,17 @@ function _handle_fetched_asset(asset: AssetEntry, url?: string) {
 }
 
 function _apply_configuration_from_args(config: MonoConfig) {
-    for (const k in (config.environment_variables || {}))
-        mono_wasm_setenv(k, config.environment_variables![k]);
+    const envars = (config.environment_variables || {});
+    if (typeof (envars) !== "object")
+        throw new Error("Expected config.environment_variables to be unset or a dictionary-style object");
+
+    for (const k in envars) {
+        const v = envars![k];
+        if (typeof (v) === "string")
+            mono_wasm_setenv(k, v);
+        else
+            throw new Error(`Expected environment variable '${k}' to be a string but it was '${v}'`);
```

Comment:
Maybe, if it seems better:

```suggestion
            throw new Error(`Expected environment variable '${k}' to be a string but it was ${typeof v}: '${v}'`);
```